app/controllers/api/v2/play_roles_controller.rb

```diff
@@ -0,0 +1,93 @@
+module Api
+  module V2
+    # API controller for Ansible Plays
+    class PlayRolesController < ::Api::V2::BaseController
+      include ::Api::Version2
+      include ForemanTasks::Triggers
+
+      rescue_from ActiveRecord::RecordNotFound, :with => :resource_not_found
+
+      before_action :find_hosts, :only => [:play_roles_on_host]
```

Comment:
If you extend hosts/hostgroups separately, you can rely on `find_resource` from core to do this (https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/architectures_controller.rb#L7)